File: java/socketd/src/main/java/org/noear/socketd/transport/client/ClientInternal.java
package org.noear.socketd.transport.client;

import org.noear.socketd.transport.core.Processor;

/**
 * 客户端内部接口
 *
 * @author noear
 * @since  2.1
 */
public interface ClientInternal extends Client {
    /**
     * 获取连接处理器
     */
    ClientConnectHandler getConnectHandler();

    /**
     * 获取心跳处理
     */
    ClientHeartbeatHandler getHeartbeatHandler();

    /**
     * 获取心跳间隔（毫秒）
     */
    long getHeartbeatInterval();

    /**
     * 获取配置
     */
    ClientConfig getConfig();

    /**
     * 获取处理器
     */
    Processor getProcessor();
}

File: go/socketd/transport/client/client_internal.go
package client

import (
	"time"

	"socketd/transport/core"
)

type ClientInternal interface {
	Client //继承

	GetHeartbeatHandler() core.HeartbeatHandler
	GetHeartbeatInterval() time.Duration
	GetConfig() *Config
	GetProcessor() core.Processor
}
